File: problems/pairs_of_songs_with_total_durations_divisible_by_60/solution.py
class Solution:
    def numPairsDivisibleBy60(self, time: List[int]) -> int:
        lookup = {}
        counter = 0
        for t in time:
            num_needed = -t % 60
            curr_remainder = t % 60
            if num_needed in lookup:
                counter += lookup[num_needed]
            lookup[curr_remainder] = lookup.get(curr_remainder, 0) + 1
        return counter

File: problems/pairs_of_songs_with_total_durations_divisible_by_60/solution.cpp
#include <unordered_map>

class Solution {
public:
    int numPairsDivisibleBy60(vector<int>& time) {
        unordered_map<int,int> lookup = {};
        int counter = 0;
        for (const int &t : time) {
            int num_needed = (60 + (-t % 60)) % 60;
            int curr_remainder = t % 60;
            if (lookup.find(num_needed) != lookup.end())
                counter += lookup[num_needed];
            if (lookup.find(curr_remainder) != lookup.end())
                lookup[curr_remainder] += 1;
            else
                lookup[curr_remainder] = 1;
        }
        return counter;
    }
};